File: 0074-search-a-2d-matrix/solution.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m=matrix.length;
        for(int i=0;i<m;i++)
        {   
            int n=matrix[i].length;
            for(int j=0;j<n;j++)
            {
                if(matrix[i][j]==target)
                {
                    return true;
                }
            }
        }
        return false;
    }
}

File: 0074-search-a-2d-matrix/solution.c
bool searchMatrix(int** mat, int m, int* n, int target) {
    int left = 0;
    int a = *n;
    int right = m * a - 1;
    while (left <= right) {
        int mid = (left + right) / 2;
        int rowidx = mid / a;
        int colidx = mid % a;
        int val = mat[rowidx][colidx];
        if (val == target) {
            return true;
        } else if (val < target) {
            left = mid + 1;
        } else
            right = mid - 1;
    }
    return false;
}
